android/intltestapp/cpp/ReplProxy.cpp

```diff
@@ -0,0 +1,195 @@
+                                                            .withRandomSeed(-1)
+                                                            .build())
+                                .withShouldRecordStats(false)
+                                .withCallback([this](vm::GCEventKind gcEventKind,
+                                                     const char *extraInfo) {
+                                    GCCallback(gcEventKind, extraInfo);
+                                })
+                                .build())
+                .withVMExperimentFlags(vm::RuntimeConfig::getDefaultVMExperimentFlags())
+                .withES6Promise(vm::RuntimeConfig::getDefaultES6Promise())
+                .withES6Proxy(vm::RuntimeConfig::getDefaultES6Proxy())
+                .withES6Symbol(vm::RuntimeConfig::getDefaultES6Symbol())
+                // .withES6Intl(true)
+                .withEnableHermesInternal(true)
+                .withEnableHermesInternalTestMethods(true)
+                .withAllowFunctionToStringWithRuntimeSource(false)
+                .build();
+
+        mRuntime = vm::Runtime::create(config);
+
+    }
+
+    static jni::local_ref<ReplProxy::jhybriddata> initHybrid(jni::alias_ref<jclass>) {
+        return makeCxxInstance();
+    }
+
+    void nativeCollect(facebook::jni::alias_ref<facebook::jni::JString> jCause) {
+        mRuntime->collect(jCause->toStdString());
+    }
+
+    std::string nativeHeapStats() {
+        std::string s;
+        llvh::raw_string_ostream os(s);
+        mRuntime->printHeapStats(os);
+        return s;
+    }
+
+    std::string nativeEvalScript(facebook::jni::alias_ref<facebook::jni::JString> jScript) {
+        std::string script = jScript->toStdString();
+        std::string response;
+        response.reserve(1024);
+
+        vm::GCScope gcScope(mRuntime.get());
+
+        vm::Handle <vm::JSObject> global = mRuntime->getGlobal();
+        auto propRes = vm::JSObject::getNamed_RJS(
+                global, mRuntime.get(), vm::Predefined::getSymbolID(vm::Predefined::eval));
+        if (propRes == vm::ExecutionStatus::EXCEPTION) {
+            mRuntime->printException(
+                    llvh::outs(), mRuntime->makeHandle(mRuntime->getThrownValue()));
+            return "error getting 'eval' from global";
+        }
+        auto evalFn = mRuntime->makeHandle<vm::Callable>(std::move(*propRes));
+
+        bool hasColors = true;
+
+        // evaluate-line.js will color the output appropriately based on the available of this property in the global object.
+        vm::HermesValue terminalTypePropName = vm::HermesValue::encodeStringValue(vm::StringPrimitive::createNoThrow(mRuntime.get(), "_replterminaltype").get());
+        vm::HermesValue terminalTypePropVal = vm::HermesValue::encodeStringValue(vm::StringPrimitive::createNoThrow(mRuntime.get(), "android").get());
+        if(vm::ExecutionStatus::EXCEPTION ==
+            global->putComputed_RJS(global,
```

Comment:
This should probably only happen once at construction instead of every time we run a line of JS.